src/dom.ts

```diff
@@ -87,9 +90,24 @@ export class FrameExecutionContext extends js.ExecutionContext {
           ${custom.join(',\n')}
         ])
       `;
-      this._injectedPromise = this._delegate.rawEvaluate(source).then(objectId => new js.JSHandle(this, 'object', objectId));
+      this._injectedScriptPromise = this._delegate.rawEvaluate(source).then(objectId => new js.JSHandle(this, 'object', objectId));
     }
-    return this._injectedPromise;
+    return this._injectedScriptPromise;
+  }
+
+  debugScript(): Promise<js.JSHandle<DebugScript> | undefined> {
+    if (!helper.isDebugMode())
+      return Promise.resolve(undefined);
+
+    if (!this._debugScriptPromise) {
+      const source = `new (${debugScriptSource.source})()`;
+      this._debugScriptPromise = this._delegate.rawEvaluate(source).then(objectId => new js.JSHandle(this, 'object', objectId)).then(async debugScript => {
```

Comment:
I think we should catch() this debugScriptPromise and return undefined, because context may disappear while we are doing this.

Reply:
Done

Reply:
Done. But I wonder if we should have done it on call sites